app/assets/javascripts/components/PipelineSampleReads.jsx

```diff
@@ -351,6 +358,25 @@ class PipelineSampleReads extends React.Component {
         </div>
       );
     }
+    sample_dropdown = (
+      <div className='dropdown-button sample-select-dropdown' data-activates='sample-list'>
+        { this.sampleInfo.name }
+        <i className="fa fa-chevron-down right"/>
+      </div>
+    )
+    sample_dropdown_content = (
+      <ul id='sample-list' className='dropdown-content sample-dropdown-content'>
+       { Object.keys(this.sample_map).map((sample_id, i) => {
```

Comment:
looks like sample_map is undefined, it breaks at this point

Reply:
It is defined in line 10 above. Can you give me an example of a URL where it breaks?

Reply:
weird just checked out the branch again, couldn't reproduce this error, works fine.

Reply:
Weitd, maybe a github synch issue. Good point though about handling edge cases -- I've added some checks to omit the dropdown if there is a problem with sample_map or if it contains only a single sample